web/satellite/src/components/accessGrants/CreateAccessModal.vue

```diff
@@ -0,0 +1,1032 @@
+                width: 100%;
+                justify-content: space-evenly;
+                padding-top: 25px;
+
+                &__learn-more-button {
+                    padding: 0 15px;
+                }
+
+                &__copy-button {
+                    width: 49% !important;
+                    margin-right: 10px;
+                }
+
+                &__download-button {
+                    width: 49% !important;
+                }
+
+                &__encrypt-button {
+                    padding: 0 15px;
+                }
+            }
+        }
+    }
+
+    ::v-deep .buckets-selection {
+        margin-left: 0;
+        height: 30px;
+        border: 1px solid #c8d3de;
+    }
+
+    ::v-deep .buckets-selection__toggle-container {
+        padding: 10px 20px;
+    }
+
+    .permissions-chevron-up {
+        @include chevron;
+
+        transform: rotate(-90deg);
+    }
+
+    .permissions-chevron-down {
+        @include chevron;
+    }
+
+    .tooltip-icon {
+        width: 14px;
+        height: 14px;
+        cursor: pointer;
+    }
+
+    .tooltip-text {
+        text-align: center;
+        font-weight: 500;
+    }
+
+    a {
+        color: #fff;
+        text-decoration: underline !important;
+        cursor: pointer;
+    }
+
+    .access-tooltip {
+        top: 52px;
+        left: 94px;
+
+        @include tooltip-container;
+
+        &:after {
+            left: 50%;
+            top: 100%;
+
+            @include tooltip-arrow;
+        }
+    }
+
+    .s3-tooltip {
+        top: 158px;
+        left: 103px;
+
+        @include tooltip-container;
+
+        &:after {
+            left: 50%;
+            top: -8%;
+            transform: rotate(180deg);
+
+            @include tooltip-arrow;
+        }
+    }
+
+    .api-tooltip {
+        top: 186px;
+        left: 78px;
+
+        @include tooltip-container;
+
+        &:after {
+            left: 50%;
+            top: -11%;
+            transform: rotate(180deg);
+
+            @include tooltip-arrow;
+        }
+    }
+
+    @media screen and (max-height: 800px) {
+
+        .access-grant {
+            padding: 50px 0 20px;
+            overflow-y: scroll;
+        }
+    }
+
+    @media screen and (max-height: 750px) {
+
+        .access-grant {
+            padding: 100px 0 20px;
+        }
+    }
+
+    @media screen and (max-height: 700px) {
+
+        .access-grant {
+            padding: 150px 0 20px;
+        }
+    }
+
+    @media screen and (max-height: 650px) {
+
+        .access-grant {
+            padding: 200px 0 20px;
+        }
+    }
+
+    @media screen and (max-height: 600px) {
+
+        .access-grant {
+            padding: 250px 0 20px;
+        }
+    }
+
+    @media screen and (max-height: 550px) {
+
+        .access-grant {
+            padding: 300px 0 20px;
+        }
+    }
+
+    @media screen and (max-height: 500px) {
+
+        .access-grant {
+            padding: 350px 0 20px;
+        }
+    }
+</style>
```

Comment:
Do you have suggestions for the best way to split up this component into smaller changes? I am not sure how to divide it.

Reply:
this may be done in later deliveries but here are my thoughts

we could try to make 4 components from one
1. Main modal component which will control flow's state (basically all the logic)
2. Permissions step where user will select all the permissions they need (I understand that it may have lots of props passed to it)
3. Encryption step (not many props)
4. Result step  (not many props)

So basically all the logic will be in first component and most of the html/css code will be split in 3 step components (operated by props)

Please tell me if I'm wrong :)